frontend/src/Cart.js

```diff
@@ -0,0 +1,178 @@
+    console.log("payload", payload);
+    try {
+      let response = await axios.delete(url, { data: payload });
+      if (response.status === 200) {
+        dispatch({
+          type: "REMOVE_FROM_CARTLIST",
+          payload: payload.itemId
+        });
+      }
+    } catch (err) {
+      console.log("err", err);
+    }
+  }
+
+  let cartdata = cartlist.map((eachitem) => {
+    let finddata = data.find((item) => item._id === eachitem.itemId);
+    if (finddata) {
+      return { ...finddata, qty: eachitem.qty };
+    }
+    return {};
+  });
+  //console.log("cartdata de", cartdata);
+  return (
+    <>
+      <h5>Cart</h5>
+
+      <div className="items">
+        {cartdata
+          .filter((item) => item.qty > 0)
+          .map((item) => {
+            return (
+              <>
+                <div
+                  class="cards-t1"
+                  style={{ width: "40%", maxWidth: "210px" }}
+                >
+                  <img class="cards-t1-img" src={item.image} alt={item.name} />
+
+                  <div
+                    class="title"
+                    style={{
+                      overflow: "hidden",
+                      textOverflow: "...",
+                      height: "20px"
+                    }}
+                  >
+                    {item.name}
+                  </div>
+                  <div class="desc">
+                    {item.inStock ? "InStock" : "Out of Stock"}
+                  </div>
+                  <div class="price">
+                    Rs:{" "}
+                    {(item.price - (item.price * item.offer) / 100).toFixed(2)}
+                  </div>
+
+                  <div className="added_to_cart">
+                    <button
+                      onClick={() => {
+                        if (item.qty === 1) {
+                          cart_delete_call(
+                            "https://ecomm-demo.utpalpati.repl.co/cart",
+                            { itemId: item._id }
+                          );
+                        } else {
+                          cart_decrease_call(
+                            "https://ecomm-demo.utpalpati.repl.co/cart",
+                            {
+                              items: { itemId: item._id, qty: item.qty - 1 }
+                            }
+                          );
+                        }
+                      }}
+                    >
+                      -
+                    </button>
+                    {item.qty}
+
+                    <button
+                      onClick={() => {
+                        cart_increase_call(
+                          "https://ecomm-demo.utpalpati.repl.co/cart",
+                          {
+                            items: { itemId: item._id, qty: item.qty + 1 }
+                          }
+                        );
+                      }}
+                    >
+                      +
+                    </button>
+                  </div>
```

Comment:
You can use the Add_to_cart_button function written in app.js file here.

Reply:
Thanks for pointing this out. I think this will significantly reduce my lines of code.